tests/scancode/test_mark_source.py

```diff
@@ -0,0 +1,51 @@
+#
+# Copyright (c) 2017 nexB Inc. and others. All rights reserved.
+# http://nexb.com and https://github.com/nexB/scancode-toolkit/
+# The ScanCode software is licensed under the Apache License version 2.0.
+# Data generated with ScanCode require an acknowledgment.
+# ScanCode is a trademark of nexB Inc.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# When you publish or redistribute any data created with ScanCode or any ScanCode
+# derivative work, you must accompany this data with the following acknowledgment:
+#
+#  Generated with ScanCode and provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+#  OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+#  ScanCode should be considered or used as legal advice. Consult an Attorney
+#  for any legal advice.
+#  ScanCode is a free software code scanning tool from nexB Inc. and others.
+#  Visit https://github.com/nexB/scancode-toolkit/ for support and download.
+
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from os import path
+
+from commoncode.testcase import FileBasedTesting
+from scancode.cli_test_utils import run_scan_click
+
+
+class TestMarkSource(FileBasedTesting):
+
+    test_data_dir = path.join(path.dirname(__file__), 'data')
+
+    def test_scan_mark_source_without_info(self):
+        test_dir = self.get_test_loc('mark_source/test')
+        expected_file = self.get_test_loc('mark_source/without_info.expected.json')
+        result_file = self.get_temp_file('.json')
+        result = run_scan_click(['--mark-source', test_dir, result_file])
+        assert open(expected_file).read() == open(result_file).read()
```

Comment:
See https://github.com/nexB/scancode-toolkit/blob/22bf88df19768dd207973ed2db91249822a0c446/tests/scancode/test_cli.py#L275 for a test example.
The `check_json_scan` thing abstracts dates and keys order: https://github.com/nexB/scancode-toolkit/blob/develop/src/scancode/cli_test_utils.py#L45